kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt

```diff
@@ -0,0 +1,374 @@
+
+    private val activityMainBinding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val fab: FloatingActionButton by lazy {
+        activityMainBinding.fab
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+
+        // Authentication with an API key or named user is required to
+        // access basemaps and other location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        // Create a map with the BasemapType topographic
+        val map = ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC)
+
+        // Set the map to be displayed in the layout's MapView
+        mapView.map = map
+
+        // Define strings for the service URL and collection id
+        // Note that the service defines the collection id which can be
+        // accessed via OgcFeatureCollectionInfo.getCollectionId()
+        val serviceUrl = "https://demo.ldproxy.net/daraa"
+        val collectionId = "TransportationGroundCrv"
+
+        // Create an OGC feature collection table from the service url and collection ID
+        ogcFeatureCollectionTable = OgcFeatureCollectionTable(serviceUrl, collectionId)
+
+        // Set the feature request mode to manual
+        // In this mode, the table must be manually populated
+        // and panning and zooming won't request features automatically
+        ogcFeatureCollectionTable.featureRequestMode =
+            ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE
+
+        ogcFeatureCollectionTable.addDoneLoadingListener {
+            if (ogcFeatureCollectionTable.loadStatus == LoadStatus.LOADED) {
+
+                // Create a feature layer and set a renderer to it to visualize the OGC API features
+                val featureLayer = FeatureLayer(ogcFeatureCollectionTable)
+                val simpleRenderer = SimpleRenderer(
+                    SimpleLineSymbol(
+                        SimpleLineSymbol.Style.SOLID,
+                        Color.MAGENTA,
+                        3f
+                    )
+                )
+                featureLayer.renderer = simpleRenderer
+
+                // Add the layer to the map
+                map.operationalLayers.add(featureLayer)
+
+                // Zoom to the dataset extent
+                val datasetExtent = ogcFeatureCollectionTable.extent
+                if (datasetExtent != null && !datasetExtent.isEmpty) {
+                    mapView.setViewpointGeometryAsync(datasetExtent)
+                }
+
+                // Create a query based on the current visible extent
+                val visibleExtentQuery = QueryParameters()
+                visibleExtentQuery.geometry = datasetExtent
+
+                // Set a limit of 3000 on the number of returned features per request,
```

Comment:
ah I see this comment here. you might want to change the initiation of this value to 3000 then